### PR TITLE
feat: create ui select showtime by province and cinema

### PR DESCRIPTION
**Pull request by TonNQ**
Show showtimes of film by selected province and cinema in detail film page

86ept9dhv

### CODE REVIEW

File: src/components/CinemaShow/CinemaShow.jsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
+import IconArrowUp from '@/assets/images/ic-arrow-up.svg'
```
Comment (ces-anhlam) [inline src/components/CinemaShow/CinemaShow.jsx:1]: import as component

File: src/components/ShowTime/ShowTime.jsx [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    document.addEventListener('mousedown', handleClickOutside)
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside)
+    }
+  }, [])
+
+  useEffect(() => {
+    dispatch(setSelectedFilm(film))
+
+    if (film?.dateList?.length > 0) {
```
Comment (ces-anhlam) [inline src/components/ShowTime/ShowTime.jsx:53]: destructuring first

File: src/components/ShowTime/ShowTime.jsx [hunk 3]
```diff
@@ -0,0 +1,126 @@
+import TimeCard from '@/components/TimeCard'
+import CinemaShow from '@/components/CinemaShow'
+import IconLocation from '@/assets/images/ic-location.svg'
```
Comment (ces-anhlam) [inline src/components/ShowTime/ShowTime.jsx:3]: import as component

File: src/utils/price.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+export const formatPrice = (price, decimalSeparator = '.', thousandSeparator = ',') => {
+  let priceStr = price.toString()
```
Comment (ces-anhlam) [inline src/utils/price.js:2]: `const`

File: src/utils/price.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
+export const formatPrice = (price, decimalSeparator = '.', thousandSeparator = ',') => {
```
Comment (ces-anhlam) [inline src/utils/price.js:1]: I don't see you use thí function on this PR, please remove that

File: src/components/CinemaShow/CinemaShow.jsx (hunk 1 above)
Comment (TonNQ) [reply src/components/CinemaShow/CinemaShow.jsx:1]: fixed

File: src/components/ShowTime/ShowTime.jsx (hunk 3 above)
Comment (TonNQ) [reply src/components/ShowTime/ShowTime.jsx:3]: fixed

File: src/components/ShowTime/ShowTime.jsx (hunk 2 above)
Comment (TonNQ) [reply src/components/ShowTime/ShowTime.jsx:53]: fixed

File: src/utils/price.js (hunk 4 above)
Comment (TonNQ) [reply src/utils/price.js]: I removed this file

File: src/utils/price.js (hunk 5 above)
Comment (TonNQ) [reply src/utils/price.js]: I removed this file

File: src/utils/datetime.js [hunk 6]
```diff
@@ -6,3 +6,40 @@ export const formatDate = (dateString, template = 'DD/MM/YYYY') => {
   }
   return null
 }
+
+export const getDayOfWeek = (dateString) => {
```
Comment (ces-anhlam) [inline src/utils/datetime.js:10]: Can you use `momentjs` to perform this function more easily?

File: src/utils/datetime.js [hunk 7]
```diff
@@ -6,3 +6,40 @@ export const formatDate = (dateString, template = 'DD/MM/YYYY') => {
...
+
+    const daysOfWeek = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday']
+
+    return daysOfWeek[date.getDay()]
+  } else {
+    return null
+  }
+}
+
+export const isCurrentTimeGreaterThan = (givenTime) => {
```
Comment (ces-anhlam) [inline src/utils/datetime.js:24]: same above

File: src/utils/datetime.js [hunk 8]
```diff
@@ -6,3 +6,40 @@ export const formatDate = (dateString, template = 'DD/MM/YYYY') => {
...
+    const givenDate = new Date()
+    givenDate.setHours(givenHours, givenMinutes, 0, 0)
+
+    return currentDate > givenDate
+  } else {
+    return null
+  }
+}
+
+export const getHourAndMinute = (time) => {
```
Comment (ces-anhlam) [inline src/utils/datetime.js:38]: same above

File: tailwind.config.js [hunk 9]
```diff
@@ -64,7 +64,6 @@ const getFilterString = (filterValue) => {
 
 const convertFiltersToUtilities = (filters) => {
   const utilities = {}
-
```
Comment (ces-anhlam) [inline tailwind.config.js]: unnecessary changed

File: src/utils/datetime.js (hunk 7 above)
Comment (TonNQ) [reply src/utils/datetime.js:24]: fixed

File: src/utils/datetime.js (hunk 6 above)
Comment (TonNQ) [reply src/utils/datetime.js:10]: I fixed all by using momentjs

File: src/utils/datetime.js (hunk 8 above)
Comment (TonNQ) [reply src/utils/datetime.js:38]: fixed

File: tailwind.config.js (hunk 9 above)
Comment (TonNQ) [reply tailwind.config.js]: fixed

File: src/utils/datetime.js [hunk 10]
```diff
@@ -6,3 +6,30 @@ export const formatDate = (dateString, template = 'DD/MM/YYYY') => {
...
+    const givenMoment = moment(givenDateTimeString, 'YYYY-MM-DD HH:mm:ss')
+
+    return currentTime.isAfter(givenMoment)
+  }
+  return null
+}
+
+export const getHourAndMinute = (time) => {
+  if (time) {
+    const momentTime = moment(time, 'HH:mm:ss')
```
Comment (ces-anhlam) [inline src/utils/datetime.js:31]:
```
return moment(time, 'HH:mm:ss').format('HH:mm')
``` is enough

File: src/utils/datetime.js [hunk 11]
```diff
@@ -6,3 +6,30 @@ export const formatDate = (dateString, template = 'DD/MM/YYYY') => {
   }
   return null
 }
+
+export const getDayOfWeek = (dateString) => {
+  if (dateString) {
+    const date = moment(dateString, 'DD/MM/YYYY')
```
Comment (ces-anhlam) [inline src/utils/datetime.js:12]: same above

File: src/utils/datetime.js (hunk 10 above)
Comment (TonNQ) [reply src/utils/datetime.js:31]: fixed

File: src/utils/datetime.js (hunk 11 above)
Comment (TonNQ) [reply src/utils/datetime.js:12]: fixed

**Review by ces-anhlam — APPROVED** [review]

**Review by ces-khanhhoang — APPROVED** [review]

**Pull request merged**